e2e-tests/testdata/echo_client/ts/client.ts

```diff
@@ -815,6 +820,7 @@ class BaseClient {
     public async callAPI(method: string, path: string, body?: BodyInit, params?: CallParameters): Promise<Response> {
         let { query, ...rest } = params ?? {}
         const init = {
+            ...this.requestInit,
             ...rest,
```

Comment:
If `rest` contains the `headers` field it will entirely overwrite the `headers` field from `requestInit`, whereas we want to merge the two